packages/api/amplify_api_dart/lib/src/graphql/web_socket/services/web_socket_service.dart

```diff
@@ -95,6 +96,24 @@ class AmplifyWebSocketService
     }
   }
 
+  /// Generates a list of protocols from a [WebSocketState].
+  @visibleForTesting
+  Future<List<String>> generateProtocols(
+    ApiOutputs outputs,
+    AmplifyAuthProviderRepository authRepo,
+  ) async {
+    final authorizationHeaders = await generateAuthorizationHeaders(
+      outputs,
+      isConnectionInit: true,
+      authRepo: authRepo,
+      body: appSyncDefaultPayload,
+    );
+    final encodedAuthHeaders = base64Url
+        .encode(json.encode(authorizationHeaders).codeUnits)
+        .replaceAll('=', '');
```

Comment:
It is safe to remove all. base64url uses `=` as padding only. Padding is optional according to the base64url spec, but dart includes it. We want to remove it since it is not allowed in a protocol name. I pushed a comment to explain that.